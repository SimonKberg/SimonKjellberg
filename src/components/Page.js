// @flow strict
import * as React from 'react'
import Head from 'next/head'
import styled from '@emotion/styled'
import { theme } from 'styled-tools'

import Header from './Header'

const Wrapper = styled('div')`
  font: 100%/1.5 Iosevka, monospace;
  color: ${theme('color.primary')};
`

const Content = styled('div')`
  margin: 0 auto;
  max-width: ${theme('page.maxWidth')};
  padding: 0 0.9375rem 1.875rem;
  padding-top: 0;
`

type Props = {
  siteTitle: string,
  siteDescription: string,
  children?: React.Node,
}

const Page = ({ children, siteTitle, siteDescription }: Props) => (
  <Wrapper>
    <Head>
      <title>{siteTitle}</title>
      <meta property="og:title" content={siteDescription} />
      <meta
        name="description"
        property="og:description"
        content={siteDescription}
      />
    </Head>
    <Header siteTitle={siteTitle} />
    <Content>{children}</Content>
  </Wrapper>
)

Page.defaultProps = {
  siteTitle: 'Simon Kjellberg',
  siteDescription: '',
}

export default Page
