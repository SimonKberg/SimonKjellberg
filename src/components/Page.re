[@bs.config {jsx: 3}];

open Utils;

module Styles = {
  open Css;

  let wrapper =
    style([
      fontSize(pct(100.0)),
      lineHeight(`abs(1.5)),
      fontFamily("Iosevka, monospace"),
      color(Theme.Color.primary),
    ]);

  let content =
    style([
      margin2(~v=`zero, ~h=`auto),
      maxWidth(Theme.Page.maxWidth),
      padding3(~top=`zero, ~h=rem(0.9375), ~bottom=rem(1.875)),
      unsafe(
        "@supports(padding: max(0px)) and (padding: env(safe-area-inset-bottom))",
        style([
          unsafe("padding-left", "max(env(safe-area-inset-left), 0.9375rem)"),
          unsafe(
            "padding-right",
            "max(env(safe-area-inset-right), 0.9375rem)",
          ),
          unsafe(
            "padding-bottom",
            "max(env(safe-area-inset-bottom), 1.875rem)",
          ),
        ]),
      ),
    ]);
};
let component = ReasonReact.statelessComponent("Link");

[@react.component]
let make = (~siteTitle="Simon Kjellberg", ~siteDescription="", ~children) => {
  <div className=Styles.wrapper>
    <Next.Head>
      <title> {siteTitle |> str} </title>
      <meta property="og:title" content=siteTitle />
      <meta
        name="description"
        property="og:description"
        content=siteDescription
      />
    </Next.Head>
    <Header siteTitle />
    <div className=Styles.content> children </div>
  </div>;
};

let default = make;
