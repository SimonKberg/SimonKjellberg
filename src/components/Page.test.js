// @flow strict
import * as React from 'react'
import renderer from 'react-test-renderer'
import Page from './Page.bs'

test('<Page />', () => {
  const tree = renderer.create(<Page />).toJSON()
  expect(tree).toMatchSnapshot()
})
