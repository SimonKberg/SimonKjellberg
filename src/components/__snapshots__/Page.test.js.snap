// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> 1`] = `
.emotion-2 {
  font: 100%/1.5 Iosevka,monospace;
}

.emotion-0 {
  margin: 0 auto;
  padding: 0 0.9375rem 1.875rem;
  padding-left: max(0.9375rem,env(safe-area-inset-left));
  padding-right: max(0.9375rem,env(safe-area-inset-left));
  padding-bottom: max(1.875rem,env(safe-area-inset-bottom));
}

<div
  className="emotion-2 emotion-3"
>
  <Header
    siteTitle="Simon Kjellberg"
  />
  <div
    className="emotion-0 emotion-1"
  />
</div>
`;
